soco/events_base.py

```diff
@@ -0,0 +1,699 @@
+            # (with a capital letter)
+            if timeout.lower() == 'infinite':
+                self.timeout = None
+            else:
+                self.timeout = int(timeout.lstrip('Second-'))
+            self._timestamp = time.time()
+            self.is_subscribed = True
+            log.info(
+                "Renewed subscription to %s, sid: %s",
+                self.service.base_url + self.service.event_subscription_url,
+                self.sid)
+
+        # pylint: disable=missing-docstring
+        def failure():
+
+            # If the renewal has failed, it seems appropriate
+            # to cancel the subscription.
+            self._cancel_subscription()
+
+            err_msg = "Could not renew subscription to {}, sid: {}".format(
+                self.service.base_url + self.service.event_subscription_url,
+                self.sid)
+            if autorenew:
+                log.warning(err_msg)
+                return
+            raise SoCoException(err_msg)
+
+        self._request(
+            'SUBSCRIBE',
+            self.service.base_url + self.service.event_subscription_url,
+            headers, success, failure)
+
+    def unsubscribe(self):
+        """unsubscribe()
+        Unsubscribe from the service's events.
+        Once unsubscribed, a Subscription instance should not be reused
+        """
+        # Trying to unsubscribe if already unsubscribed, or not yet
+        # subscribed, fails silently
+        if self._has_been_unsubscribed or not self.is_subscribed:
+            return
+        self._cancel_subscription()
+
+        # If the subscription has timed out, an attempt to
+        # unsubscribe from it will fail.
+        if self.time_left == 0:
+            log.info(
+                "Subscription %s, sid: %s has expired. No need to unsubscribe",
+                self.service.base_url + self.service.event_subscription_url,
+                self.sid)
+            return
+
+        # Send an unsubscribe request like this:
+        # UNSUBSCRIBE publisher path HTTP/1.1
+        # HOST: publisher host:publisher port
+        # SID: uuid:subscription UUID
+        headers = {
+            'SID': self.sid
+        }
+
+        # pylint: disable=missing-docstring, unused-argument
+        def success(*arg):
+            log.info(
+                "Unsubscribed from %s, sid: %s",
+                self.service.base_url + self.service.event_subscription_url,
+                self.sid)
+
+        # pylint: disable=missing-docstring
+        def failure():
+            log.warning(
+                "Error attempting to unsubscribe from %s, sid: %s",
+                self.service.base_url + self.service.event_subscription_url,
+                self.sid)
+
+        self._request(
+            'UNSUBSCRIBE',
+            self.service.base_url + self.service.event_subscription_url,
+            headers, success, failure)
+
+    def send_to_service(self, event):
```

Comment:
Is there a reason why this method is called `send_to_service`, it doesn't use the service? I would have suggested `send_event` or something similiar.